lapce-data/src/document.rs

```diff
@@ -1364,20 +1365,31 @@ impl Document {
         self.apply_deltas(&deltas)
     }
 
-    pub fn styles(&self) -> Option<&Arc<Spans<Style>>> {
-        let styles = self
-            .semantic_styles
-            .as_ref()
-            .or_else(|| self.syntax().and_then(|s| s.styles.as_ref()));
-        styles
+    pub fn styles(&self) -> Option<Arc<Spans<Style>>> {
+        let semantic_styles = self.semantic_styles.as_ref();
+        let syntax_styles = self.syntax().and_then(|s| s.styles.as_ref());
+
+        if semantic_styles.is_some() && syntax_styles.is_some() {
```

Comment:
You should do `match (semantic_styles, syntactic_styles)` here, I think and handle the 4 (maybe 3 if you can merge the two middle cases into an or pattern) branches in the match.

Reply:
thanks! that looks a lot better.